src/libraries/System.Private.CoreLib/src/System/Single.cs

```diff
@@ -153,21 +163,22 @@ internal static uint ExtractTrailingSignificandFromBits(uint bits)
         }
 
         /// <summary>Determines whether the specified value is finite (zero, subnormal, or normal).</summary>
+        /// <remarks>This effectively checks the value is not NaN and not infinite.</remarks>
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsFinite(float f)
         {
-            int bits = BitConverter.SingleToInt32Bits(f);
-            return (bits & 0x7FFFFFFF) < 0x7F800000;
+            uint bits = BitConverter.SingleToUInt32Bits(f);
+            return (~bits & PositiveInfinityBits) != 0;
```

Comment:
Is this more efficient, or it's just clearer what it's doing?

Reply:
Bit more efficient (both perf and space wise) for `64-bit`, while allowing smaller code in hoistable loops for 32-bit. It also allows branching code to be a bit more efficient (instead of `and; cmp; jcc`, we just do `andn; jcc`).

Basically the old pattern was:
```x86asm
; 32-bit
vmovd eax, xmm0                     ;  4-bytes, 3-cycles
and   eax, 0x7FFF_FFFF              ;  5-bytes, 1-cycles
cmp   eax, 0x7F80_0000              ;  5-bytes, 1-cycles
setl  al                            ;  3-bytes, 1-cycles
movzx eax, al                       ;  3-bytes, 1-cycles
                                    ; 20-bytes, 7-cycles

; 64-bit
vmovd rax, xmm0                     ;  5-bytes, 3-cycles
mov   rcx, 0x7FFF_FFFF_FFFF_FFFF    ; 10-bytes, 1-cycles
and   rax, rcx                      ;  3-bytes, 1-cycles
mov   rcx, 0x7FF0_0000_0000_0000    ; 10-bytes, 1-cycles
cmp   rax, rcx                      ;  3-bytes, 1-cycles
setl  al                            ;  3-bytes, 1-cycles
movzx eax, al                       ;  3-bytes, 1-cycles
                                    ; 37-bytes, 9-cycles
```

While the new pattern can be much smaller:
```x86asm
; 32-bit
vmovd eax, xmm0                     ;  4-bytes, 3-cycles
mov   ecx, 0x7F80_0000              ;  5-bytes, 1-cycles
andn  eax, eax, ecx                 ;  5-bytes, 1-cycles
setne al                            ;  3-bytes, 1-cycles
movzx rax, al                       ;  3-bytes, 1-cycles
                                    ; 20-bytes, 7-cycles

; 64-bit
vmovd rax, xmm0                     ;  5-bytes, 3-cycles
mov   rcx, 0x7FF0_0000_0000_0000    ; 10-bytes, 1-cycles
andn  rax, rax, rcx                 ;  5-bytes, 1-cycles
setne al                            ;  3-bytes, 1-cycles
movzx rax, al                       ;  3-bytes, 1-cycles
                                    ; 26-bytes, 7-cycles
```